sql/core/src/main/scala/org/apache/spark/sql/execution/SparkSqlParser.scala

```diff
@@ -884,6 +884,10 @@ class SparkSqlAstBuilder(conf: SQLConf) extends AstBuilder {
     // to include the partition columns here explicitly
     val schema = cols ++ partitionCols
 
+    // Ensuring whether no duplicate name is used in table definition;
+    // Also ensuring the existing columns are not used as partition columns
+    checkDuplicateNames(colNames = schema.map(_.name), ctx)
```

Comment:
After code changes, we are verifying two cases: one is duplicate names in table definition; another is column repeated in partitioning columns.